PINFuture/Classes/PINFuture.m

```diff
@@ -50,7 +50,7 @@ - (instancetype)initPrivate
     return self;
 }
 
-- (void)setValue:(id)value
+- (void)setValue:(nullable id)value
```

Comment:
The intention behind this was do be `notnull`.  If it's notnull, do you remember what the problem is?

Reply:
if we change the value to be nullable we also have to change the setter to be nullable.

This assumption was made based on if you look into:
```
- (void)transitionToState:(PINFutureState)state value:(nullable NSObject *)value error:(nullable NSError *)error
```
You will see that the value can be nullable that will be assigned to the `value` property.